src/Microsoft.Identity.Web.TokenAcquisition/MergedOptions.cs

```diff
@@ -426,7 +427,7 @@ internal static void ParseAuthorityIfNecessary(MergedOptions mergedOptions)
                 if (indexTenant >= 0)
                 {
                     mergedOptions.Instance = authority.Substring(0, indexTenant);
-                    mergedOptions.TenantId = authority.Substring(indexTenant + 1);
+                    mergedOptions.TenantId = mergedOptions.PreserveAuthority ? null : authority.Substring(indexTenant + 1);
```

Comment:
I'm not sure about this. Let's take a CIAM CUD authority

`https://cats.com/12345/v2.0/`

then the `Instance` would be `https://cats.com/` and the `TenantId` would be null.

Where is "Instance" used elsewhere in the code?

Reply:
In this code and after this method is called, `Instance` is stored in the same field of `confidentialClientApplicationOptions`.

Reply:
For CIAM, just like for B2C, customers will use `"authority"` not `Instance` and `TenantId`. CIAM would not use `Instance`

Reply:
I am trying to understand the logic of PreserveAuthority exactly, it seems to imply that the authority is not Ciam if set to true right? In that case, wouldnt we want to use the TenantId from the authority like we do in the original code?

Why wouldnt this line be 

`mergedOptions.PreserveAuthority ? authority.Substring(indexTenant + 1) : null;`

Reply:
Ok, then at least let's add a comment to "Instance". 

IMO, if a property isn't supposed to be used / trusted, there should be some protection against it. I'm worried we will forget about this convetion and we will use "Instance" later in some other part of the code that will blow up.

Reply:
I'm reopening this. `Instance` is used today in managing the dictionary of "CCA" objects, it's part of the key there. 

So there could be a potential conflict once we start supporting other generic authorities, for example: 

`https://demo.duendesoftware.com/A` vs `https://demo.duendesoftware.com/B` would both have the same instance and "Tenant" = null. 

IMO, if preseve authority = true then: 

- instance = authority
- tenant = null